apptools/undo/tests/test_command_stack.py

```diff
@@ -0,0 +1,119 @@
+# -----------------------------------------------------------------------------
+#
+#  Copyright (c) 2015, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in enthought/LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+# -----------------------------------------------------------------------------
+
+from traits.testing.unittest_tools import unittest
+
+from apptools.undo.api import CommandStack, UndoManager
+from testing_commands import SimpleCommand, UnnamedCommand
+
+
+class TestCommandStack(unittest.TestCase):
+    def setUp(self):
+        self.stack = CommandStack()
+        undo_manager = UndoManager()
+        self.stack.undo_manager = undo_manager
+
+        self.command = SimpleCommand()
+
+    # Assertion helpers -------------------------------------------------------
+
+    def assert_n_commands_pushed(self, n):
+        self.assertEqual(len(self.stack._stack), n)
+        self.assertEqual(self.stack._index, n-1)
+        self.assertFalse(self.stack.clean)
+
+    def assert_n_commands_pushed_and_undone(self, n):
+        # N commands have been pushed
+        self.assertEqual(len(self.stack._stack), n)
+        self.assertEqual(self.stack._index, -1)
```

Comment:
I'm sure there's a good reason for this, but it deserves a comment to clarify what's going on.